src/Symfony/Component/TypeInfo/TypeContext/TypeContextFactory.php

```diff
@@ -188,4 +184,59 @@ private function collectTemplates(\ReflectionClass|\ReflectionFunctionAbstract $
 
         return $templates;
     }
+
+    /**
+     * @return array<string, Type>
+     */
+    private function collectTypeAliases(\ReflectionClass $reflection, TypeContext $typeContext): array
+    {
+        if (!$this->stringTypeResolver || !class_exists(PhpDocParser::class)) {
+            return [];
+        }
+
+        if (!$rawDocNode = $reflection->getDocComment()) {
+            return [];
+        }
+
+        $aliases = [];
+        foreach ($this->getPhpDocNode($rawDocNode)->getTagsByName('@phpstan-type') + $this->getPhpDocNode($rawDocNode)->getTagsByName('@psalm-type') as $tag) {
+            if (!$tag->value instanceof TypeAliasTagValueNode) {
+                continue;
+            }
+
+            $aliases[$tag->value->alias] = $this->stringTypeResolver->resolve((string) $tag->value->type, $typeContext);
+        }
+
+        foreach ($this->getPhpDocNode($rawDocNode)->getTagsByName('@phpstan-import-type') + $this->getPhpDocNode($rawDocNode)->getTagsByName('@psalm-import-type') as $tag) {
```

Comment:
with the `+` doesn't this mean that the phpstan has priority if conflict ? would it be worth noting this for the documentation or setting this to two variable and checking for intersect keys ?

Reply:
IMO, having either `@phpstan-*` and `@psalm` is really uncommon, if it does happen, the developer should "expect" to have collisions.